src/main/java/com/nifcloud/mbaas/core/NCMBInstallation.kt

```diff
@@ -486,51 +486,78 @@ class NCMBInstallation : NCMBObject {
          * @return NCMBInstallation object that is created from data that is saved to local file.<br></br>
          * If local file is not available, it returns empty NCMBInstallation object
          */
-        fun getCurrentInstallation(): NCMBInstallation {
-            //null check
-            checkNCMBContext()
-            try {
-                //create currentInstallation
-                if (installation == null) {
-                    installation = NCMBInstallation()
-                    //ローカルファイルに配信端末情報があれば取得、なければ新規作成
-                    val currentInstallationFile = create(INSTALLATION_FILENAME)
-                    if (currentInstallationFile.exists()) {
-                        //ローカルファイルから端末情報を取得
-                        val localData = readFile(currentInstallationFile)
-                        installation = NCMBInstallation(localData)
+        var currentInstallation: NCMBInstallation? = null
+            get() {
+                //null check
+                checkNCMBContext()
+                try {
+                    //create currentInstallation
+                    if (field == null) {
+                        field = NCMBInstallation()
+                        //ローカルファイルに配信端末情報があれば取得、なければ新規作成
+                        val currentInstallationFile = create(INSTALLATION_FILENAME)
+                        if (currentInstallationFile.exists()) {
+                            //ローカルファイルから端末情報を取得
+                            val localData = readFile(currentInstallationFile)
+                            field = NCMBInstallation(localData)
+                        }
                     }
+                } catch (error: NCMBException) {
+                    throw NCMBException(error)
                 }
-            } catch (error: Exception) {
-                Log.e("Error", error.toString())
+                return field as NCMBInstallation
```

Comment:
こちらの型変換は必要ないかと思いますが、いかがでしょうか
```suggestion
                return field
```

Reply:
削除しました。